### ISSUE TITLE
Ingredient Notes Rework

### ISSUE THREAD

**Issue by hotdog218** [vabene1111/recipes#455]
**Is your feature request related to a problem? Please describe.**
It's kind of a pain when cooking to hover over the comment icon to see the note. 

**Describe the solution you'd like**
It would be cool to have notes that are under x amount of characters just display text. For example, if you just want to specify "softened" for butter, then it would just show up as "butter, _softened_". But if you want to say something longer, like "Make sure the bananas are very ripe", then the current hover system would be used.

**Describe alternatives you've considered**
Text wrapping could also be a viable solution. Maybe a user setting that will toggle between hovering and wrapping.

**Additional context**
Mostly a QoL feature, but for me, the minimal amount of interaction I can make with the computer while cooking, the better.

Example:
![3f2b49b541d166d17e2f24751382448e](https://user-images.githubusercontent.com/24951087/108925631-3cf69980-75fa-11eb-8483-1b30061d7798.png)

**Comment by vabene1111** [comment]
this is a cool idea, i have a big mix of long and short notes as well .. it coudl also be a setting i guess 🤔 definitely something i want to do

**Comment by Szeraax** [comment]
I'd suggest only implementing this or #661.

**Comment by vabene1111** [comment]
i will look at both of those options when i get to this, thanks for the link

**Comment by Vishrut19** [comment]
@smilerz Do you need any help or assistance ? I can help you out if you are interested

**Comment by vabene1111** [comment]
Feel free to give it a try, it's not super complicated but the solution needs to be well thought trough.

**Comment by Vishrut19** [comment]
Hi @vabene1111 I will try to solve it and also I have joined the discord server.Hope you and other people will help me and support. Since this is my first open source contribution please help me out if I need any help.

**Comment by vabene1111** [comment]
Sure, feel free to ask. Also please share a rough draft of what you want to do before implementing anything.

**Comment by xmayukx** [comment]
Hello, @hotdog218 I can work on this issue, could you please assign it to me?

**Comment by hotdog218** [comment]
> Hello, @hotdog218 I can work on this issue, could you please assign it to me?

I don't believe I have the ability to assign anything, plus I think @pedro-camargo11 is already handling this- thanks by the way!

**Comment by smilerz** [comment]
> Hello, @hotdog218 I can work on this issue, could you please assign it to me?

No need to have it assigned to you.  If you have thoughts on how to address you can describe your solution here. Or submit a PR and it’ll be reviewed for merge.

**Comment by Royal-Dragon** [comment]
is this still open for contribution?

**Comment by vabene1111** [comment]
somebody opened a PR here https://github.com/TandoorRecipes/recipes/pull/2643

**Comment by aditya-jnu** [comment]
Hey I would like to work on this can I get started

**Comment by hotdog218** [comment]
There's already an open PR for this, but I suppose if you wanted to make it a setting that is still something that could be done given that the PR has been open for a while.

**Comment by aditya-jnu** [comment]
Actually this is my first time trying to do an open source not having any
prior idea. Can you please suggest what should I do?

**Comment by smilerz** [comment]
> Actually this is my first time trying to do an open source not having any prior idea. Can you please suggest what should I do?

do you know python or vue?

**Comment by aditya-jnu** [comment]
I am sorry to say tha I know none of the two.

**Comment by smilerz** [comment]
> I am sorry to say tha I know none of the two.

Contributing to this project probably isn't for you then.

**Comment by aditya-jnu** [comment]
Ok, thanks anyway.

**Comment by JamesOsborn-SE** [comment]
I'd argue that the view page should not be a table but a list or a two column table with the green check mark(I don't know what it's for) and the ingredient with the note separated by a comma. It's standard on the recipe sites I frequent. 

like so:
(ignore the missing checkmark)

![image](https://github.com/TandoorRecipes/recipes/assets/3580335/9283c09b-0b2d-41fd-8f63-fc4ccfda68ac)


with something like this (I can optimize this later):
```Typescript

        displayIngredient: function(){
            let result = ""
            if(this.amount)
                result += this.amount
            if(this.unitName)
                result += " "+ this.unitName 
            result += " " + this.foodName 
            if(this.ingredient.note)
                result += ", " + this.ingredient.note
            return result
        }
```

Let me know if you feel this is out of scope for this ticket. It would be much more usable for me if it was formatted this way. 

I can make it a setting and allow a toggle if that works for you.

**Comment by vabene1111** [comment]
thanks for your feedback. Please do not submit any PR as the frontend is currently being rebuild. I can imagine having a setting for something like this in the future.

**Comment by NickstaDB** [comment]
Sounds like the UI rebuild is well underway but having deployed Tandoor myself recently and running into the same issue I thought might solution might be of interest - I've just placed the ingredient note in "muted" text below the ingredient itself.

![image](https://github.com/user-attachments/assets/aad356f3-bce2-49a8-9b29-9114993314f6)

Diff of the code changes here: https://github.com/NickstaDB/recipes/commit/dae114fcae8723c362c0ad4210e31aea41ecc6d8

**Comment by ZD-FLoM** [comment]
Earlier comment suggests #661 as a possible alternative, show ingredient notes as a 5th column. This sounds like it would clutter up phone screen sizes though, but on a tablet or PC monitor that sounds fine. 

 @NickstaDB 
Growing the row for an ingredient and showing the description there looks great and would fit smaller screen sizes also, maybe even 1 or 2 font sizes smaller. Ideally font size would be adjustable, but I think that is where the custom CSS theme file comes in. 

My own experience when reading a recipe is with a screen that I can't touch because I am cooking. Having to click a note to read more details slows me down and gets the screen messy.

**Comment by lewisburgess** [comment]
I (using Claude Code!) created a pull request https://github.com/TandoorRecipes/recipes/pull/4508 that allows you to toggle ingredients notes between the original bubble tooltip and in-line:

<img width="513" height="91" alt="Image" src="https://github.com/user-attachments/assets/dbbe3168-7414-4b4c-a15d-0e09a8e64710" />

**Comment by BolishettiAkshay** [comment]
Hi, I’m new to open-source contributions and would like to work on this issue. I noticed that there have been previous PRs and that the frontend is being rebuilt, so I wanted to check if this issue is still open for contribution.

My idea is to implement a simple approach where short ingredient notes (under a certain character limit) are displayed inline (e.g., “butter, softened”), while longer notes continue to use the existing tooltip behavior. If preferred, I could also explore making this a toggle/setting.

Please let me know if this aligns with the current direction of the project or if there’s a better way to approach it.

**Comment by smilerz** [comment]
> Hi, I’m new to open-source contributions and would like to work on this issue. I noticed that there have been previous PRs and that the frontend is being rebuilt, so I wanted to check if this issue is still open for contribution.

This has already been built and is waiting for a merge.  #4601

**Comment by Shubham-21-coder** [comment]
#assign me ,i have a great experience to solve such type of problem

**Comment by DivyaKPatel** [comment]
Hi! I’d be interested in working on this issue.

<img width="1536" height="1024" alt="Image" src="https://github.com/user-attachments/assets/1fb06430-9031-4853-9630-9315bb92ee8e" />

My suggestion would be to display notes inline using a comma, for example: butter, softened or bananas, very ripe. It feels natural, keeps the ingredient list readable, and avoids requiring users to hover while cooking.

Longer notes could wrap naturally on smaller screens, while the existing tooltip could remain available as an optional display preference.

I’ve also mocked up a few alternative approaches for comparison. Would this direction work for the project? I’d be happy to implement it.
